core/src/modules/ops/destination.ts

```diff
@@ -912,13 +913,45 @@ export namespace DestinationOps {
     success: boolean;
     error: Error;
   }> {
-    const _exports: Export[] = []; // only ones we are sending
-    for (const _export of givenExports) {
-      // TODO: maybe should recalcuate hasChanges and from current mostRecent
-      if (!_export.hasChanges) {
-        await _export.complete(); // do not include exports with hasChanges=false
+    const _exports: Export[] = [];
+    const locks: Lock[] = [];
+
+    for (const givenExport of givenExports) {
+      const lock = await getOrSetLock({
+        key: `${givenExport.recordId}:${givenExport.destinationId}`,
+      });
+      locks.push(lock);
+
+      if (lock.isLocked) {
+        // is _export the newest export for this pair?  if not, cancel it.
+        const mostRecentExport = await Export.findOne({
```

Comment:
Note this could, and mostly commonly would, return `givenExport`.
Maybe it should say `not` that one and then check for null.